tracer/build/_build/docker/smoke.windows.nuget.dockerfile

```
@@ -0,0 +1,49 @@
+﻿ARG DOTNETSDK_VERSION
+ARG RUNTIME_IMAGE
+
+# Build the ASP.NET Core app using the latest SDK
+FROM mcr.microsoft.com/dotnet/sdk:$DOTNETSDK_VERSION as builder
+
+# Without this, the container doesn't have permission to add the new package 
+USER ContainerAdministrator
+
+# Build the smoke test app
+WORKDIR /src
+COPY ./test/test-applications/regression/AspNetCoreSmokeTest/ .
+
+ARG PUBLISH_FRAMEWORK
+RUN dotnet restore "AspNetCoreSmokeTest.csproj" \
+    && dotnet nuget add source "c:\src\artifacts" \
+    && dotnet add package "Datadog.Monitoring.Distribution" --prerelease \
```

If we add the source flag to the [dotnet add package command](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-add-package#options), would that guarantee we're using the artifacts of the local build instead of nuget.org?

```suggestion
    && dotnet add package "Datadog.Monitoring.Distribution" --prerelease --source "c:\src\artifacts" \
```

---

_Sigh_ you'd have thought so, wouldn't you. After all, that's what we do in the _linux_ version of this. However, long, painful iteration in CI has shown me that you _can't_ do this here apparently, as otherwise it tries to source _all_ the NuGet packages from this location (e.g. `System.*` packages), which fails obviously. This is the only incantation I could find that worked 🤷